spices/SPICE-0013-bytes-standard-library.adoc

```diff
@@ -0,0 +1,376 @@
+= Bytes standard library class
+
+* Proposal: link:./SPICE-0013-bytes-standard-library.adoc[SPICE-0013]
+* Author: https://github.com/bioball[Dan Chao]
+* Status: TBD
+* Implemented in: TBD
+* Category: Standard Library
+
+== Introduction
+
+This SPICE introduces a new value type called `Bytes`, which represents binary data.
+
+== Motivation
+
+Pkl currently does not have a binary data type.
+
+This introduces the following limitations:
+
+* `pkl eval` can only produce UTF-8 encoded strings.
+* Asides from values coming from `read()`, it is not possible to compute checksums of binary content.
+* It is harder to interop with configuration formats that allow binary data.
+
+== Proposed Solution
+
+A new class will be added to `pkl:base`, called `Bytes`.
+
+== Detailed design
+
+=== Standard library changes
+
+The following changes will be made to the standard library
+
+==== `Bytes` class
+
+A new class will be added to `pkl:base`, called `Bytes`.
+This will be a new value type, similar to `Duration` and `DataSize`.
+
+[source,pkl]
+----
+@Since { version = "0.29.0" }
+external class Bytes extends Any {
+  /// The value of these bytes.
+  external value: List<UInt8>
+
```

Comment:
I see reference to `toList()` below. What's the difference between `toList()` and `value`?

Also, would it make sense to add a `length` property?

Reply:
Ah, I went back and forth on these two names, and forgot to update this.

I think `toList()` is probably better here; more consistent with the rest of our API.

Reply:
I actually went back on this; I'm keeping this as `value`. This better captures the idea that `Bytes` is just a wrapper, and has an underlying value.

Reply:
Update: this got changed back again; `toList()` is back. See https://github.com/apple/pkl-evolution/blob/509c11a8b678f3d6f7bd637cbd4456dba107e5d3/spices/SPICE-0013-bytes-standard-library.adoc#bytes-as-a-wrapper for rationale